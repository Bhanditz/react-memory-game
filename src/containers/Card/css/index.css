.react-card-flip {
  display: block;
  height: 215px; }
  .react-card-flip .panel {
    position: relative;
    border: 1px solid #eeeeee;
    border-radius: 10px;
    padding: 15px;
    width: 200px;
    height: 200px; }
    .react-card-flip .panel img {
      max-height: 100%;
      max-width: 100%;
      width: auto;
      height: auto;
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      right: 0;
      margin: auto; }

/*# sourceMappingURL=index.css.map */
