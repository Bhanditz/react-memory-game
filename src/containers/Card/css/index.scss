.react-card-flip {
  display: block;
  height: 215px;
  .panel {
    position: relative;
    border: 1px solid #eeeeee;
    border-radius: 10px;
    padding: 15px;
    width: 200px;
    height: 200px;

    img {
      max-height: 100%;
      max-width: 100%;
      width: auto;
      height: auto;
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      right: 0;
      margin: auto;
    }
  }

}

