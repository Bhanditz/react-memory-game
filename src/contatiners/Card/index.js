import React, {Component} from 'react';
import ReactCardFlip from 'react-card-flip'
import {Image} from 'react-bootstrap'
import cardBack from './images/back.jpg';
import {CARD_FLIP_SPEED} from '../../config'
import PropTypes from 'prop-types';

const styles = {
    card: {
        border: '1px solid #eeeeee',
        borderRadius: '3px',
        padding: '15px',
        width: '250px'
    }
};

class Card extends Component {

    handleClick() {
        const {model, onClick} = this.props;

        if (model.isFlipped) {
            onClick(model.id);
        }

    }

    render() {
        const {model} = this.props;

        return (
            <ReactCardFlip
                isFlipped={model.isFlipped}
                flipSpeedBackToFront={CARD_FLIP_SPEED}
                flipSpeedFrontToBack={CARD_FLIP_SPEED}>

                <div key="front" onClick={this.handleClick.bind(this)} style={styles.card}>
                    <Image src={model.image}/>
                </div>

                <div key="back" onClick={this.handleClick.bind(this)} style={styles.card}>
                    <Image src={cardBack}/>
                </div>
            </ReactCardFlip>
        );
    }
}

Card.propTypes = {
    onClick: PropTypes.func.isRequired,
    model: PropTypes.shape({
        matched: PropTypes.bool.isRequired,
        isFlipped: PropTypes.bool.isRequired,
        id: PropTypes.string.isRequired,
        image: PropTypes.string.isRequired,
    })
};

export default Card;
