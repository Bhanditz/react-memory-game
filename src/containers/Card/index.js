import React, {Component} from 'react';
import {Panel, Image} from 'react-bootstrap'
import PropTypes from 'prop-types';
import "./css/index.css"
import FlippingCard from '../../components/FlippingCard';

class Card extends Component {

    handleClick() {
        const {model, onClick} = this.props;

        if (model.isFlipped) {
            onClick(model.id);
        }

    }

    render() {
        const {model} = this.props;

        return (

            <div className="card-holder">
                <FlippingCard
                    isFlipped={model.isFlipped}
                    Front={<Panel key="front" onClick={this.handleClick.bind(this)}><Image src={model.image}/>
                    </Panel>}
                    Back={<Panel key="back" onClick={this.handleClick.bind(this)}/>}
                />
            </div>

        );
    }
}

Card.propTypes = {
    onClick: PropTypes.func.isRequired,
    model: PropTypes.shape({
        matched: PropTypes.bool.isRequired,
        isFlipped: PropTypes.bool.isRequired,
        id: PropTypes.string.isRequired,
        image: PropTypes.string.isRequired,
    })
};

export default Card;
