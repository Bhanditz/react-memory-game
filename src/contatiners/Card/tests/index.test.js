import React from 'react';
import Card from '../index'
import renderer from 'react-test-renderer';

describe('Card', () => {

    const onClick = () => {
        console.log();
    };
    const model = {
        matched: false,
        isFlipped: false,
        id: "fake",
        image: "http://myimg.com"
    };

    it('should match snapshot', () => {
        const tree = renderer.create(<Card onClick={onClick} model={model}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

});