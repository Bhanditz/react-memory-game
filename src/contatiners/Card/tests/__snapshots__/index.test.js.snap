// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<div
  className="react-card-flip"
  style={
    Object {
      "perspective": "1000px",
      "transformStyle": "preserve-3d",
    }
  }
>
  <div
    className="react-card-flipper"
    style={
      Object {
        "position": "relative",
        "transform": "rotateY(0deg)",
        "transformStyle": "preserve-3d",
        "transition": "0.6s",
      }
    }
  >
    <div
      className="react-card-front"
      style={
        Object {
          "WebkitBackfaceVisibility": "hidden",
          "backfaceVisibility": "hidden",
          "left": "0",
          "position": "absolute",
          "top": "0",
          "transform": "rotateY(0deg)",
          "transformStyle": "preserve-3d",
          "width": "100%",
          "zIndex": "2",
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "border": "1px solid #eeeeee",
            "borderRadius": "3px",
            "padding": "15px",
            "width": "250px",
          }
        }
      >
        <img
          className=""
          src="http://myimg.com"
        />
      </div>
    </div>
    <div
      className="react-card-back"
      style={
        Object {
          "WebkitBackfaceVisibility": "hidden",
          "backfaceVisibility": "hidden",
          "left": "0",
          "position": "absolute",
          "top": "0",
          "transform": "rotateY(180deg)",
          "transformStyle": "preserve-3d",
          "width": "100%",
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "border": "1px solid #eeeeee",
            "borderRadius": "3px",
            "padding": "15px",
            "width": "250px",
          }
        }
      >
        <img
          className=""
          src="back.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
